examples/todolist/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#[derive(Debug, Clone)]
+struct TodoEntry {
+    text: String,
+}
+
+// I am a simple Todolist
+#[derive(Debug, Clone)]
+struct TodoList {
+    items: Vec<String>,
+}
+
+impl TodoList {
+    fn new() -> Self {
+        Self { items: Vec::new() }
+    }
+
+    fn add_item<S: Into<String>>(&mut self, item: S) {
+        self.items.push(item.into())
+    }
+
+    fn get_last(&self) -> String {
+        self.items.last().cloned().unwrap()
+    }
+
+    fn add_entry(&mut self, entry: TodoEntry) {
+        self.items.push(entry.text)
+    }
```

Comment:
Could you please also add a `get_last_entry` method that returns the entire `TodoEntry` record, and call it from the tests? This will be a nice way to exercise the foreign language support for lifting from a buffer rather than a raw pointer, which IIUC is not currently covered.